### DISCUSSION TITLE
Why does `RequestHeaders` have the `get` key?

### DISCUSSION THREAD

**Discussion by aarontravass** [graphql/graphql-http#99] [Q&A] [answered]
Servers implementing the `IncomingHttpHeaders` interface result in conflicting types. How can it be resolved?

**Comment by enisdenjo** [comment]
The [Fetch API request headers use the get method](https://developer.mozilla.org/en-US/docs/Web/API/Headers/get) while [Node's incoming message headers is an object](https://nodejs.org/api/http.html#messageheaders).

There should be no conflicts, can you share an example or a repro?

**Reply by aarontravass** [reply]
something like this
```ts
import { createHandler } from "graphql-http";

const handler = createHandler({
  context: (req) => ({
    auth: req.headers.authorization
  }),
});

```
Error:
![image](https://github.com/graphql/graphql-http/assets/43901677/21709327-ee86-49dd-b1cc-0724f6f91102)

Conflict:
```ts
import type { Request } from "@tinyhttp/app";
import {createHandler} from "graphql-http";

const handler = createHandler({
  context: (req: Request) => ({
      auth: req.headers.authorization
    })
  });

```
![image](https://github.com/graphql/graphql-http/assets/43901677/5015bdfe-0117-4f5b-bee0-963c93768a0b)

**Reply by enisdenjo** [reply, marked as answer]
[`graphql-http` has its own Request](https://github.com/graphql/graphql-http/blob/main/docs/interfaces/handler.Request.md), you cannot override it. You can either:

```ts
import { createHandler } from 'graphql-http';

const handler = createHandler({
  context: (req) => ({
    auth:
      typeof req.headers.get === 'function'
        ? req.headers.get('authorization')
        : Object(req.headers).authorization,
  }),
});
```

or

```ts
import { createHandler } from 'graphql-http';
import type { Request } from '@tinyhttp/app';

const handler = createHandler<Request>({
  context: (req) => ({
    auth: req.raw.headers.authorization,
  }),
});
```

**Reply by aarontravass** [reply]
But shouldn't `RequestHeaders` implement the same interface as `IncomingHttpHeaders` from "http"?

**Reply by enisdenjo** [reply]
It does implement it.

`graphql-http` is server agnostic, it can run in any JavaScript environment - both the [Fetch API request](https://developer.mozilla.org/en-US/docs/Web/API/Headers/get) and [Node's `http` module](https://nodejs.org/api/http.html#messageheaders). So it implements both:

https://github.com/graphql/graphql-http/blob/91aea45415180107c6b6cc0fca94a7a57f4b2d77/src/handler.ts#L31-L47
